File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums)
        heap = [(-fr,num) for num, fr in freq.items()]
        heapq.heapify(heap)
        ans = [heapq.heappop(heap)[1] for _ in range(k)]
        return ans

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>> pq;
        unordered_map<int, int> cnt;
        vector<int> ans;
        
        for (auto n: nums)
            cnt[n]++;
        for (auto [k, v]: cnt)
            pq.emplace(v, k);
        
        while (k--) {
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};